vault/request_handling.go

```diff
@@ -1007,6 +995,23 @@ func (c *Core) handleRequest(ctx context.Context, req *logical.Request) (retResp
 		if rawVals, ok := entry.synthesizedConfigCache.Load("audit_non_hmac_request_keys"); ok {
 			nonHMACReqDataKeys = rawVals.([]string)
 		}
+
+		// Allowing writing to a path ending in / makes it extremely difficult to
```

Comment:
Why did this move into handleRequest? I believe now we will do this check/error out quite a bit later in the process and this won't apply to login mounts (which is okay but still a little weird).

Reply:
Oh, probably needs to be in both, but the reason is we don't have the mount entry in handleCancellableRequest.